### DISCUSSION TITLE
Quick help needed for missing Tasmota modul

### DISCUSSION THREAD

**Discussion by Pewibe** [arendst/Tasmota#15014] [Q&A] [answered]
I compiled Tasmota by myself because it was needed to activate SML. Month later I recognized that a device (Sonoff Basic R3) running this firmware is not able to operate as access point. After entering this mode (7 times power toggle) my configuration is lost but no network "tasmota_XXXXXX-XXXX" is on air. 
Is there someone who knows which line(s) I messed up in my_user_config.h, user_config_override.h ?
I could not recognize a #DEFINE like INCLUDE_ACCESS_POINT_CAPABILITY or something similar.
Thanks in advance!

**Comment by Jason2866** [comment]
We dont know your config so we cant know what you may have messed up.
In general there is no #define for (de)activating AP mode. This does not mean you have choosen settings which prevents Power Recovery mode. If you cant recover your device (see FAQ in Docu). Serial flashing is the way.

**Reply by Pewibe** [reply]
Thanks @Jason2866. I hoped there is such a general #define. And recovering is not my problem because I edited  user_config_override.h with my WLAN parameters. Problem is that I helped several friends with such devices to enable them to remotely read their Smart Meters. These guys are no insides but need the AP to enter their SSID / PW.
I attached my_user_config.h (renamed) to enable you to have a look on it.
[my_user_config.txt](https://github.com/arendst/Tasmota/files/8170522/my_user_config.txt)
.

**Comment by Jason2866** [comment]
With a fast view over the config i see nothing which prevents recovery. BUT every setting done after the initial config is persistent. So everything can be done...

**Comment by Pewibe** [comment]
@Jason2866, if you know any command to configurate which leads to the described behavior (7 times power toggle works, the actual config is scratched out, but no network "tasmota_XXXXXX-XXXX" on air), I feel glad to hear your advices. 
BTW, these are my used commands after initial flash until problem (just 2 commands per cut-n-paste):
```
Template {"NAME":"Sonoff Basic","GPIO":[17, 1, 1, 1, 1, 1, 1, 1,224,288, 1, 1, 1 ,1],"FLAG":0,"BASE":18}
Backlog Savedata 1; 
  Hostname H-StromZaehler-106; IpAddress1 192.168.0.106; 
  DeviceName H.StromZaehler.106;
  MqttHost 192.168.0.12; MqttClient H.StromZaehler.106; Topic H.StromZaehler;
  FriendlyName .106; 
  StateText1 ---; StateText2 ===;
  WebButton1 SEND-MQTT-TEST;
  WebColor13 #505050; WebColor14 #303030; WebColor19 #1fa3ec;
  SO3 0; SO56 1; SO57 1; Timezone 99; NtpServer1 192.168.0.1;  
  PowerOnState 0; LedPower 1; LedState 1; LedMask 0x0004;
Savedata 0;
```
How about a bug? Should I try the same with the newest version?

**Reply by barbudor** [reply]
>if you know any command to configurate which leads to the described behavior (7 times power toggle works, the actual config is scratched out, but no network "tasmota_XXXXXX-XXXX" on air)

You shouldn't make any change in `user_config.h` but only do override settings in `user_config_override.h`
This makes much more easy to see what you have changed vs the official configuration

With `WIFI_CONFIG_TOOL       WIFI_RETRY`, if Tasmota is not switching back to AP, I would expect that it is because your flashed binary has a default AP configured.

As you are likely to have to connect to serial port to reflash, I would suggest you take the opportunity first to see what is exactly happenning with a terminal software.

**Reply by Pewibe** [reply]
Thanks @barbudor for your help. Concerning user_config.h I know that it is easier if I do no change. But modules are getting bigger with Smart Meter Interface and comment in .h-file says "the second section contains Enabled and Disabled FEATURES allowing different program size" I tried to cut down the overall size. 

With "your flashed binary has a default AP configured" you mean that I should define empty STA_SSID1 / STA_PASS1 instead of my Wifi parameters? I had that idea, too and thought of testing this. If your expectation is true, I think the comment in this definition should warn like "defining SSID do not allow to use Access Point mode of Tasmota"!

**Comment by Pewibe** [comment]
As @barbudor suggested I figured out (by several builds of the .bin file) that the problem is the definition of "YourSSID" and "YourPassword" in user_config_override.h. Restore these overrides to the original source code the "Fast Power Cycle Device Recovery" works as described. I like to propose these 2 steps:
1. The author of the .h file should add comment that "Fast Power Cycle Device Recovery" is stripped down to config reset, but do not provide an AP. 
2. The ".../docs/Device-Recovery" chapter or the ".../docs/Getting-Started/#initial-configuration" chapter (which one is correct?) should explain clearly that the appearance of an AP depends on unchanged user_config_override.h.

**Reply by ascillato** [reply]
In the default configuration, WIFICONFIG is set to 4 (RETRY). With this configuration, it checks for the SSID1 and SSID2 to connect to. If they are empty, this parameter will turn on the AP mode. If you have something in SSID1 in the default configuration, Tasmota will try to connect to it instead.

**Comment by Jason2866** [comment, marked as answer]
Fast Power Recycle resets to defaults. So if defaults are changed in `user_config_override.h` this one will be used. If the defaults are vaild wifi settings the are used. Thats why it is there.
